packages/core/actions/account-client/prepareMultiUserOpRequest.ts

```diff
@@ -0,0 +1,239 @@
+import {
+    AccountOrClientNotFoundError,
+    type UserOperation,
+    estimateUserOperationGas,
+    parseAccount
+} from "permissionless"
+import type {
+    PrepareUserOperationRequestParameters,
+    PrepareUserOperationRequestReturnType,
+    SponsorUserOperationReturnType
+} from "permissionless/actions/smartAccount"
+import type { StateOverrides } from "permissionless/types/bundler"
+import type {
+    ENTRYPOINT_ADDRESS_V07_TYPE,
+    EntryPoint,
+    GetEntryPointVersion
+} from "permissionless/types/entrypoint"
+import type {
+    Abi,
+    Address,
+    Chain,
+    Client,
+    Hex,
+    LocalAccount,
+    Transport
+} from "viem"
+import { estimateFeesPerGas, getChainId } from "viem/actions"
+import type { Prettify } from "viem/chains"
+import { type EncodeDeployDataParameters, getAction } from "viem/utils"
+import type { KernelSmartAccount } from "../../accounts"
+import { getMultiUserOpDummySignature } from "../../accounts/kernel/utils/plugins/multi-chain/getMultiUserOpDummySignature.js"
+
+export type SmartAccount<
+    entryPoint extends EntryPoint,
+    TSource extends string = string,
+    transport extends Transport = Transport,
+    chain extends Chain | undefined = Chain | undefined,
+    TAbi extends Abi | readonly unknown[] = Abi
+> = LocalAccount<TSource> & {
+    client: Client<transport, chain>
+    entryPoint: entryPoint
+    getNonce: () => Promise<bigint>
+    getInitCode: () => Promise<Hex>
+    getFactory: () => Promise<Address | undefined>
+    getFactoryData: () => Promise<Hex | undefined>
+    encodeCallData: (
+        args:
+            | {
+                  to: Address
+                  value: bigint
+                  data: Hex
+              }
+            | {
+                  to: Address
+                  value: bigint
+                  data: Hex
+              }[]
+    ) => Promise<Hex>
+    getDummySignature(
+        userOperation: UserOperation<GetEntryPointVersion<entryPoint>>
+    ): Promise<Hex>
+    encodeDeployCallData: ({
+        abi,
+        args,
+        bytecode
+    }: EncodeDeployDataParameters<TAbi>) => Promise<Hex>
+    signUserOperation: (
+        userOperation: UserOperation<GetEntryPointVersion<entryPoint>>
+    ) => Promise<Hex>
+}
+
+export async function prepareMultiUserOpRequest<
```

Comment:
Can move this action to multi chain package as well along with `getMultiUserOpDummySignature` util